File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Tasks/managingtaskcosts.rb
module Asposetasksjava
  module ManagingTaskCosts
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new

        task = project.getRootTask().getChildren().add("Task")

        tsk = Rjb::import('com.aspose.tasks.Tsk')

    	task.set(tsk.COST, Rjb::import('java.math.BigDecimal').valueOf(800))

    	puts "Task Remaining Cost: " + task.get(tsk.REMAINING_COST).to_string
    	puts "Task Fixed Cost: " + task.get(tsk.FIXED_COST).to_string
    	puts "Task Cost Variance: " + task.get(tsk.COST_VARIANCE).to_string
    	puts "Project Cost: " + project.getRootTask().get(tsk.COST).to_string
    	puts "Project Fixed Cost: " + project.getRootTask().get(tsk.FIXED_COST).to_string
    	puts "Project Remaining Cost: " + project.getRootTask().get(tsk.REMAINING_COST).to_string
    	puts "Project Variance Cost: " + project.getRootTask().get(tsk.COST_VARIANCE).to_string
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithTasks/ManagingTaskCosts.php
<?php
namespace Aspose\Tasks\WorkingWithTasks;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\Tsk as Tsk;
use java\math\BigDecimal as BigDecimal;

class ManagingTaskCosts{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project();

        $task = $project->getRootTask()->getChildren()->add("Task");

        $tsk = new Tsk();

        $bigDecimal=new BigDecimal();
    	$task->set($tsk->COST, $bigDecimal->valueOf(800));

    	print "Task Remaining Cost: " . (string)$task->get($tsk->REMAINING_COST).PHP_EOL;//.to_string
        print "Task Fixed Cost: " . (string)$task->get($tsk->FIXED_COST).PHP_EOL;//.to_string
        print "Task Cost Variance: " . (string)$task->get($tsk->COST_VARIANCE).PHP_EOL;//.to_string
        print "Project Cost: " . (string)$project->getRootTask()->get($tsk->COST).PHP_EOL;//.to_string
        print "Project Fixed Cost: " . (string)$project->getRootTask()->get($tsk->FIXED_COST).PHP_EOL;//.to_string
        print "Project Remaining Cost: " . (string)$project->getRootTask()->get($tsk->REMAINING_COST).PHP_EOL;//.to_string
        print "Project Variance Cost: " . (string)$project->getRootTask()->get($tsk->COST_VARIANCE).PHP_EOL;//.to_string
    }

}